Luke/DateRangePicker_Update

lukeschunk opened the pull request:
There's a new design for the `<DateRangePicker />` and this makes those updates. 

Some Highlights: 
* Adds "DateSelectBox"  - these are now part of the `<SelectionPane />` component 
* Adds an `onClose` prop. This will be useful for taking action (ie fetching data) when the date picker is closed
* Factored half the component (the left half on desktop) to a `<SelectionPane />`
* Updated Default Ranges as per design request 
* Adds `<YearSelector />` and `<MonthSelector />` components to calendar to improve UX (ie: less clicks to go back a year)
* Some minor refactoring to make things more discoverable

paniclater commented on src/components/DateRangePicker/SelectionPane.js:
```diff
@@ -0,0 +1,128 @@
+  };
+
+  state = {
+    selectedBox: 'from'
```
What are the other options this can be set to? Should this be enumerated in a constants file or object?

lukeschunk replied:
done

paniclater commented on src/components/DateRangePicker/SelectionPane.js:
```diff
@@ -0,0 +1,128 @@
+  }
+
+  _handleDateBoxClick = (date, selectedBox) => {
+    this.props.handleDateBoxClick(date, selectedBox);
```
I believe we settled on using the `onDoSomething` style for the prop name and the imperative `doSomething` name for the function passed in for prop handlers. Correct me if I'm wrong though, I'm always mixing those up.

paniclater commented on src/components/DateRangePicker/SelectionPane.js:
```diff
@@ -0,0 +1,128 @@
+      <div style={styles.container}>
+        <div style={{ borderBottom: isLargeOrMediumWindowSize ? 'none' : '1px solid ' + StyleConstants.Colors.FOG }}>
+          <label style={styles.boxLabel}>From</label>
+          <div onClick={() => this._handleDateBoxClick(selectedStartDate, 'from')} style={Object.assign({}, styles.dateSelectBox, this.props.selectedBox === 'from' ? styles.selectedDateSelectBox : null)}>{selectedStartDate ? moment.unix(selectedStartDate).format('MMM D, YYYY') : 'Select Start Date'}</div>
```
Yeah, feels like 'from' should maybe be a constant here.

lukeschunk replied:
what would the constant object structure look like?

lukeschunk replied:
i guess i'll just do a mirrored object with `FROM: 'FROM'`

paniclater commented on src/components/DateRangePicker/SelectionPane.js:
```diff
@@ -0,0 +1,128 @@
+  }
+
+  styles = () => {
+    const { isLargeOrMediumWindowSize } = this.props;
```
I think including the isLargeOrMediumWindowSize here directly instead of passing it in via props will make this component more portable.

paniclater commented on src/components/DateRangePicker/SelectionPane.js:
```diff
@@ -0,0 +1,128 @@
+        boxSizing: 'border-box',
+        color: StyleConstants.Colors.CHARCOAL,
+        display: 'flex',
+        flexWrap: 'wrap',
```
Does using `wrap` here make more sense than having them laid out in rows? Do they need to dynamically wrap?

lukeschunk replied:
the component that the `rangeOptions` style is applied to maps over the ranges. if i did it in rows wouldn't i need to set a new element for every two elements in the ranges? this is nice because no matter the number of items in the range it works nicely: 

<img width="625" alt="screen shot 2017-07-18 at 7 10 42 pm" src="https://user-images.githubusercontent.com/13579578/28346235-dea9261a-6bec-11e7-9f67-1cccf3d0c397.png">


if you're thinking of something i'm not though lmk

paniclater commented on src/components/DateRangePicker/Selector.js:
```diff
@@ -0,0 +1,116 @@
+  }
+}
+
+class MonthSelector extends React.Component {
```
Is there a reason MonthSelector & YearSelector can't be configured into a single component?

lukeschunk replied:
no reason, but I think that this is clearer & more semantic in the parent component https://github.com/mxenabled/mx-react-components/pull/635/files#diff-b5e02cb776f0bf4505e16b8ba45bcf8dR257

paniclater commented on src/components/DateRangePicker/SelectionPane.js:
```diff
@@ -0,0 +1,128 @@
+        fontSize: StyleConstants.FontSizes.MEDIUM,
+        color: StyleConstants.Colors.CHARCOAL
+      },
+      dateSelectBox: {
```
what is the difference between `dateSelectBox` and `selectedDateSelectBox`?

lukeschunk replied:
i think the gif will make this more clear, but the `selectedDateSelectBox` is styling for when the component has focus

lukeschunk commented:
thanks for the review @paniclater also sorry i thought i edited to add the gifs but here they are: 

### Desktop 
![desktop _gif](https://user-images.githubusercontent.com/13579578/28345012-7e3f2796-6be4-11e7-8826-fd1fafa5a8fc.gif)

### Mobile
![mobile_ gif](https://user-images.githubusercontent.com/13579578/28345013-7e45937e-6be4-11e7-8c21-743df8ed6c8d.gif)

lukeschunk commented:
@paniclater this is ready for another look. 

* Changed the 2nd DateRangePicker to be called `DateRangePickerTest` 
* adjusted prop names
* used constant for `TO` and `FROM` 
* fixed CSS so spacing between two panes is equal: 

![screen shot 2017-07-21 at 11 31 49 am](https://user-images.githubusercontent.com/13579578/28478291-0d814aa6-6e15-11e7-911a-a4b8c9c615aa.png)

paniclater commented:
:shipit: thanks!
